src/Psl/Type/Internal/NonEmptyDictType.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Type\Internal;
+
+use Psl;
+use Psl\Str;
+use Psl\Type;
+use Psl\Type\Exception\AssertException;
+use Psl\Type\Exception\CoercionException;
+
+use function is_array;
+use function is_iterable;
+
+/**
+ * @template Tk of array-key
+ * @template Tv
+ *
+ * @extends Type\Type<non-empty-array<Tk, Tv>>
+ *
+ * @internal
+ */
+final class NonEmptyDictType extends Type\Type
+{
+    /**
+     * @param Type\TypeInterface<Tk> $key_type
+     * @param Type\TypeInterface<Tv> $value_type
+     *
+     * @throws Psl\Exception\InvariantViolationException If $key_value, or $value_type is optional.
+     */
+    public function __construct(
+        private Type\TypeInterface $key_type,
+        private Type\TypeInterface $value_type
+    ) {
+        Psl\invariant(
+            !$key_type->isOptional() && !$value_type->isOptional(),
+            'Optional type must be the outermost.'
+        );
+    }
+
+    /**
+     * @throws CoercionException
+     *
+     * @return non-empty-array<Tk, Tv>
+     */
+    public function coerce(mixed $value): array
+    {
+        if (is_iterable($value)) {
+            $key_trace   = $this->getTrace()
+                ->withFrame(Str\format('non-empty-array<%s, _>', $this->key_type->toString()));
```

Comment:
can we use also use dict instead of array here? :) thanks.

Reply:
oops!